master/internal/db/postgres.go

```diff
@@ -147,15 +147,23 @@ func (db *PgDB) rawQuery(q string, args ...interface{}) ([]byte, error) {
 	return ret, nil
 }
 
-// QueryDB query the database and ignore the returned rows.
-func (db *PgDB) QueryDB(queryName string, args ...interface{}) error {
-	query, ok := db.queries[queryName]
-	if !ok {
-		query = string(etc.MustStaticFile(fmt.Sprintf("%s.sql", queryName)))
-		db.queries[queryName] = query
+// ArchiveExperiment sets the experiment archived flag.
+func (db *PgDB) ArchiveExperiment(experimentID int, isArchived bool) error {
+	query := `
+UPDATE experiments
+SET archived = $2
+WHERE id = $1
+	`
+	res, err := db.sql.Exec(query, experimentID, isArchived)
+	if err != nil {
+		return errors.Wrap(err, "failed to execute query to set archived status")
+	}
+	if numRows, rowsErr := res.RowsAffected(); rowsErr != nil {
+		return errors.Wrap(rowsErr, "checking affected rows for saving experiment archived status")
```

Comment:
do we want archiving to be idempotent?

Reply:
yep, I think so (wdy think?). It should be idempotent right now. I added a check and to avoid issuing a database update if it's already archived